sky/spot/spot_state.py

```diff
@@ -419,9 +470,13 @@ def set_cancelling(job_id: int):
             (SpotStatus.CANCELLING.value, time.time(), job_id))
         if rows.rowcount > 0:
             logger.info('Cancelling the job...')
+    if callback_func is not None:
```

Comment:
This might need to go within the condition `rows.rowcount > 0` above, as this function will be called even if the job is already failed. We only need to call the event callback when the job is actually being cancelled. (same for the `set_cancelled` function)